tests/cogs/downloader/test_git.py

```diff
@@ -326,16 +330,65 @@ async def test_git_get_full_sha1_from_ambiguous_commits(git_repo):
     )
     assert p.returncode == 128
     assert p.stderr.decode().strip() == (
-        "error: short SHA1 95da0b57 is ambiguous\n"
+        "error: short object ID 95da0b57 is ambiguous\n"
         "hint: The candidates are:\n"
         "hint:   95da0b576 commit 2019-10-22 - Ambiguous commit 16955\n"
         "hint:   95da0b57a commit 2019-10-22 - Ambiguous commit 44414\n"
         "fatal: Needed a single revision"
     )
 
 
+@pytest.mark.skipif(
+    GIT_VERSION < (2, 31), reason="This is test for output from Git 2.31 and newer."
+)
 @pytest.mark.asyncio
 async def test_git_get_full_sha1_from_ambiguous_tag_and_commit(git_repo):
+    # 2 ambiguous refs:
+    # branch ambiguous_with_tag - c6f0e5ec04d99bdf8c6c78ff20d66d286eecb3ea
+    # tag ambiguous_tag_66387 - c6f0e5ec04d99bdf8c6c78ff20d66d286eecb3ea
+    p = await git_repo._run(
+        ProcessFormatter().format(
+            git_repo.GIT_GET_FULL_SHA1, path=git_repo.folder_path, rev="c6f0"
+        )
+    )
+    assert p.returncode == 128
+    assert p.stderr.decode().strip() == (
+        "error: short object ID c6f0 is ambiguous\n"
+        "hint: The candidates are:\n"
+        "hint:   c6f028f tag ambiguous_tag_66387\n"
+        "hint:   c6f0e5e commit 2019-10-24 - Commit ambiguous with tag.\n"
+        "fatal: Needed a single revision"
+    )
+
+
+@pytest.mark.skipif(
+    GIT_VERSION >= (2, 31), reason="This is test for output from Git older than 2.31."
+)
+@pytest.mark.asyncio
+async def test_git_get_full_sha1_from_ambiguous_commits(git_repo):
```

Comment:
Doesn't this need a different name?

Reply:
Yes, you're right. I just updated name of this test, thanks!